Added libsql adapter

abdel-17 opened the pull request:
All tests pass on the local database and on my remote Turso db.

pilcrowOnPaper commented:
Hey, I'll be holding off reviewing and merging major PRs until I finish v2. Will help with the migration too.

abdel-17 commented:
That’s totally fine. I’m excited for the new v2 release. I tried to make the adapter work as much as possible with the existing codebase, so hopefully the migration process should be smooth.

pilcrowOnPaper commented:
The update should simplify the adapter significantly, and as a bonus, it'll make them future proof too!

abdel-17 commented:
Hey, I just updated the adapter to work with v2. The Typescript libSQL client doesn't support prepared statements yet, so I replaced those with regular `execute` function calls. I checked and it's on their roadmap, so maybe you can fix that when it drops/

taymoork2 commented:
@abdel-17 can you provide an example of how to connect this locally and to a remote turso instance (like you've mentioned above). I'm also using Drizzle, so examples for hooking up the adapter to the local or remote db using Prisma and/or Drizzle would a nice to have for updating the documentation later

@pilcrowOnPaper Any blockers on having this merged in?

abdel-17 commented:
The adapter does not really care about the ORM you use or how you connect to the db. Just pass a libSQL `Client` instance to the adapter and it should work.

I have an example [here](https://github.com/abdel-17/lucia-libsql-adapter) in my repo. You can use it until the pr gets merged.

deadcoder0904 commented:
@taymoork2 you can use [my project too as a reference point](https://github.com/deadcoder0904/next-13-lucia-auth-drizzle-turso-sqlite-magic-link/tree/main) which uses next 13, lucia v2, turso with this libsql adapter.

make sure to check the main branch only. other branches are a bit buggy. nothing to do with lucia though.

pilcrowOnPaper commented:
Can you run `pnpm auri add` in the root of the project? It should create a new changeset:

```md
---
package: "@lucia-auth/adapter-sqlite"
type: "minor"
---
```

abdel-17 commented:
@pilcrowOnPaper Done

pilcrowOnPaper commented on .auri/$t9fgucw6.md:
```diff
@@ -0,0 +1,4 @@
+---
+package: "" # package name
```
```suggestion
package: "@lucia-auth/adapter-sqlite" # package name
```

pilcrowOnPaper commented on .auri/$t9fgucw6.md:
```diff
@@ -0,0 +1,4 @@
+---
+package: "" # package name
+type: "" # "major", "minor", "patch"
```
```suggestion
type: "minor" # "major", "minor", "patch"
```

taymoork2 commented:
@abdel-17 didnt notice that it was published to npm 😄 
quick question, am i able to alter the table name away from `auth_user` to just `user` so i can make it work with my existing schema? I remember reading in another discussion that custom table names where up to the adapters????

pilcrowOnPaper commented:
Thanks!
